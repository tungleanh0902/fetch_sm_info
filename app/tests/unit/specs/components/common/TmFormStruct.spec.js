import { mount } from "@vue/test-utils"
import TmFormStruct from "common/TmFormStruct"

describe(`TmFormStruct`, () => {
  let wrapper

  const propsData = {
    width: `narrow`,
    submit: function () {},
  }

  beforeEach(() => {
    wrapper = mount(TmFormStruct, {
      propsData,
      slots: {
        default: `<div class="fake-fields"></div>`,
        footer: `<div class="fake-footer">Fake Footer</div>`,
        title: `<div class="fake-title">Fake Title</div>`,
        subtitle: `<div class="fake-subtitle">Fake Subtitle</div>`,
      },
    })
  })

  it(`has a width from props`, () => {
    expect(wrapper.vm.width).toBe(`narrow`)
  })

  it(`has a submit function`, () => {
    expect(wrapper.vm.submit).toBeTruthy()
  })

  it(`has the expected html structure`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`has a default slot`, () => {
    expect(wrapper.findAll(`.fake-fields`).length).toBe(1)
  })

  it(`has a footer slot`, () => {
    expect(wrapper.find(`.fake-footer`).text().trim()).toBe(`Fake Footer`)
  })

  it(`has a title slot`, () => {
    expect(wrapper.find(`.fake-title`).text().trim()).toBe(`Fake Title`)
  })

  it(`has a subtitle slot`, () => {
    expect(wrapper.find(`.fake-subtitle`).text().trim()).toBe(`Fake Subtitle`)
  })
})
