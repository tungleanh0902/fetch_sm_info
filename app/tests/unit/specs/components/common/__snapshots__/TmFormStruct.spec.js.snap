// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmFormStruct has the expected html structure 1`] = `
<form
  class="form"
>
   
  <header
    class="form-header"
  >
    <div
      class="form-title"
    >
      <div
        class="fake-title"
      >
        Fake Title
      </div>
    </div>
     
    <div
      class="form-subtitle"
    >
      <div
        class="fake-subtitle"
      >
        Fake Subtitle
      </div>
    </div>
  </header>
   
  <main
    class="form-main"
  >
    <div
      class="fake-fields"
    />
  </main>
   
  <footer
    class="form-footer"
  >
    <div
      class="fake-footer"
    >
      Fake Footer
    </div>
  </footer>
</form>
`;
